frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/sc-node-list.tsx

```diff
@@ -57,24 +46,37 @@ const getRows: GetRows = ({ componentProps, customData }) => {
     };
   });
 
-  if (setNodes && !_.isEqual(filteredData, nodes)) {
-    setNodes(filteredData);
-  }
-
   return rows;
 };
 
-const AttachedDevicesNodeTable: React.FC<NodeTableProps> = (props) => (
-  <div className="ceph-node-list__max-height ceph-ocs-install__node-list">
-    <Table
-      aria-label="Node Table"
-      data-test-id="attached-devices-nodes-table"
-      {...props}
-      Rows={getRows}
-      Header={getColumns}
-      virtualize={false}
-    />
-  </div>
-);
+const AttachedDevicesNodeTable: React.FC<NodeTableProps> = (props) => {
+  const { data, customData } = props;
+  const { filteredNodes, setNodes } = customData;
+  const tableData: NodeKind[] = data.filter(
+    (node: NodeKind) =>
+      filteredNodes.includes(getName(node)) ||
+      filteredNodes.includes(node.metadata.labels?.['kubernetes.io/hostname']),
+  );
+
+  React.useEffect(() => {
+    if (setNodes) {
```

Comment:
```suggestion
    if (filteredNodes) {
```
Shouldnt this be `filteredNodes` ?

Reply:
No, because Neha has used this table at several places, `setNodes` is not always there.

Reply:
See this is used here , and without "setNodes" prop
https://github.com/openshift/console/blob/28edbc8e7cf2aaa3602673cb484f8f48d7e4f579/frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/create-sc/wizard-pages/donut-chart.tsx#L144